packages/expo/src/dom/www-types.ts

```diff
@@ -10,3 +12,40 @@ export type BridgeMessage<TData extends JSONValue> = {
   type: string;
   data: TData;
 };
+
+export type MarshalPropValuePrimitiveType = number | string | boolean | null | undefined;
+export type MarshalPropValueNativeAction = (...args: any[]) => any;
+export type MarshalPropValueType =
+  | MarshalPropValuePrimitiveType
+  | MarshalPropValuePrimitiveType[]
+  | Record<string, MarshalPropValuePrimitiveType>
+  | MarshalPropValueNativeAction;
+
+export { WebViewProps };
+
+/**
+ * The explicit props for DOM components.
+ */
+interface DOMComponentProps {
+  /**
+   * The props passing to WebView.
+   */
+  dom?: Omit<WebViewProps, 'source'>;
+
+  /**
+   * Whether to resize the native WebView size based on the DOM content size.
+   * @default false
+   */
+  autoSize?: boolean;
+}
+
+/**
+ * The public props for DOM components.
+ */
+export interface DOMProps extends DOMComponentProps {
```

Comment:
Should have a generic like `DOMProps<T extends { [key: string]: MarshalPropValueType | DOMComponentProps[keyof DOMComponentProps] }>` so users can still have autocomplete for their props.